cli/cluster/cmd/replicaset.go

```diff
@@ -317,7 +314,66 @@ func AddRole(uri *url.URL, ctx RolesAddCtx) error {
 
 	source := replicaset.NewCConfigSource(collector, publisher,
 		replicaset.KeyPicker(pickPatchKey))
-	err = source.AddRole(replicaset.RolesAddCtx{
+	err = source.AddRole(replicaset.RolesUpdateCtx{
+		InstName:       ctx.InstName,
+		GroupName:      ctx.GroupName,
+		ReplicasetName: ctx.ReplicasetName,
+		IsGlobal:       ctx.IsGlobal,
+		RoleName:       ctx.RoleName,
+		Force:          ctx.Force,
+	})
+	if err == nil {
+		log.Info("Done.")
+	}
+	return err
+}
+
+// RolesRemoveCtx describes the context to remove role.
+type RolesRemoveCtx struct {
```

Comment:
If this structure is the same as `RolesAddCtx`, than is will be better to rename `RolesAddCtx` to a common/single `RolesChangeCtx` instead of using both.